src/top.rs

```diff
@@ -0,0 +1,124 @@
+use crate::config;
+use prettytable::{format, Table};
+use structopt::StructOpt;
+use url::Url;
+use vector_api_client::{
+    gql::{HealthQueryExt, TopologyQueryExt},
+    Client,
+};
+
+trait StatsWriter {
+    fn kb(&mut self, n: f64) -> String;
+}
+
+struct HumanWriter {
+    f: human_format::Formatter,
+}
+
+impl HumanWriter {
+    fn new() -> Self {
+        Self {
+            f: human_format::Formatter::new(),
+        }
+    }
+}
+
+impl StatsWriter for HumanWriter {
+    fn kb(&mut self, n: f64) -> String {
+        self.f.with_decimals(2).format(n)
+    }
+}
+
+struct LocaleWriter {
+    buf: num_format::Buffer,
+}
+
+impl LocaleWriter {
+    fn new() -> Self {
+        Self {
+            buf: num_format::Buffer::new(),
+        }
+    }
+}
+
+impl StatsWriter for LocaleWriter {
+    fn kb(&mut self, n: f64) -> String {
+        self.buf
+            .write_formatted(&(n as i64), &num_format::Locale::en);
+        self.buf.to_string()
+    }
+}
+
+fn new_formatter(humanize: bool) -> Box<dyn StatsWriter> {
```

Comment:
![image](https://user-images.githubusercontent.com/130903/96153348-725bdf80-0ec2-11eb-9067-20e64e9e53e6.png)

I am a bit curious how I can get `1.5` of an event?

Reply:
😺 Yeah, this is a bit odd. It'll be cleaned up when I do a 'humanize' sweep; I don't think `-h` needs to do anything if the value is < 1k in this case. Also, the type of `events_processed` is `MetricValue::Counter { value: f64 }`, which should probably be cast to an int in GraphQL schema to avoid having to treat this as a float downstream.